tests/test_config_consistency.py

```diff
@@ -415,3 +421,116 @@ def test_integration_enabled_true(self):
             assert list(
                 filter(lambda span: "kafka.produce" in span.get("name"), spans)
             ), f"No kafka.produce span found in trace: {spans}"
+
+
+@rfc("https://docs.google.com/document/d/1kI-gTAKghfcwI7YzKhqRv2ExUstcHqADIWA4-TZ387o/edit#heading=h.8v16cioi7qxp")
+@scenarios.tracing_config_nondefault
+@features.tracing_configuration_consistency
+class Test_Config_LogInjection_Enabled:
+    """Verify behavior of integrations automatic spans"""
+
+    def setup_log_injection_enabled(self):
+        self.message = "msg"
+        self.r = weblog.get("/log/library", params={"msg": self.message})
+
+    def test_log_injection_enabled(self):
+        assert self.r.status_code == 200
+        pattern = r'"dd":\{[^}]*\}'
+        stdout.assert_presence(pattern)
+        dd = parse_log_injection_message(self.message)
+        print(55, dd)
+        required_fields = ["trace_id", "span_id", "service", "version", "env"]
+        for field in required_fields:
+            assert field in dd, f"Missing field: {field}"
+        return
+
+
+@rfc("https://docs.google.com/document/d/1kI-gTAKghfcwI7YzKhqRv2ExUstcHqADIWA4-TZ387o/edit#heading=h.8v16cioi7qxp")
+@scenarios.tracing_config_nondefault_2
+@features.tracing_configuration_consistency
+class Test_Config_LogInjection_Default:
+    def setup_log_injection_default(self):
+        self.message = "msg"
+        self.r = weblog.get("/log/library", params={"msg": self.message})
+
+    def test_log_injection_default(self):
+        assert self.r.status_code == 200
+        pattern = r'"dd":\{[^}]*\}'
+        stdout.assert_absence(pattern)
```

Comment:
We should avoid negative tests. This test could pass due to an unexpected format or the feature is not implemented. 

cc: @cbeauchesne, @bm1549: This is an example of a negative test. Here we want to verify that log injection is disabled when the DD_LOG_INJECTION_ENABLED environment variable is not set. If a library does not support log injection at all then this test would still pass and show up as an easy win. 

Should this test exist? I think it still provides value even if it could be misinterpreted in the FPD.

Reply:
is there anything (telemetry, logs) that shows that the tracers has disabled log injection ?